pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -775,16 +834,19 @@ func (nc *Controller) monitorNodeHealth() error {
 			zoneToNodeConditions[utilnode.GetZoneKey(node)] = append(zoneToNodeConditions[utilnode.GetZoneKey(node)], currentReadyCondition)
 		}
 
-		nodeHealthData := nc.nodeHealthMap.getDeepCopy(node.Name)
-		if nodeHealthData == nil {
-			klog.Errorf("Skipping %v node processing: health data doesn't exist.", node.Name)
-			continue
-		}
 		if currentReadyCondition != nil {
+			pods, err := nc.getPodsAssignedToNode(node.Name)
+			if err != nil {
+				utilruntime.HandleError(fmt.Errorf("unable to list pods of node %v: %v", node.Name, err))
+				nc.nodesToRetry.Store(node.Name, struct{}{})
```

Comment:
Why do we need this one? We will enter the "currentReadyCondition != nil" in the next iteration anyway, right?

Reply:
Please add a comment - if I'm asking about it it means it's not obvious. Which means people will be asking about it in the future (or will simply remove it).

Reply:
Comment added.
I also added `currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue` guard so we ensure that `MarkPodsNotReady` will be called only once after status transition.